### PR TITLE
ReImplement SMEM to RMEM Based on `SwizzledLayout`

### PR DESCRIPTION
**Pull request by KuangjuX**
Draft, don't review.

### CODE REVIEW

File: include/cell/copy/warp.hpp [hunk 1]
```diff
@@ -397,9 +396,12 @@ struct SharedOffsetHelper<WarpLayout_, WarpShape_, Shared_, kMode_,
     constexpr static int kTilePerRow = Shared::kRows / WarpShape::kRows;
     constexpr static int kTilePerCol = Shared::kCols / WarpShape::kCols;
 
+    // constexpr static int kRowStride = kTilePerRow / tl::num_rows<WarpLayout>;
+    // constexpr static int kColStride =
+    //     kTilePerCol / tl::num_cols<WarpLayout> * kTilePerRow;
```
Comment (haruhi55) [inline include/cell/copy/warp.hpp:401]: Are lines 399 ~ 401 no longer useful, thus can be deleted?

File: tests/cpp/cell/test_s2r_copy.cu [hunk 2]
```diff
@@ -212,8 +217,10 @@ TEST(TestReg2Shared, operand_C_half) {
     using WarpLayout = tl::RowMajor<1, 1>;
     const int kThreads = tl::get_numel<WarpLayout> * 32;
 
-    using Shared = SharedTile<Element, tl::RowMajor<16, 16>>;
-    using Reg = RegTile<BaseTileRowMajor<Element>, tl::RowMajor<1, 1>>;
+    // using Shared = SharedTile<Element, tl::RowMajor<16, 16>>;
+    using Shared = SharedTile<Element, tl::RowMajor<64, 128>>;
+    // using Reg = RegTile<BaseTileRowMajor<Element>, tl::RowMajor<1, 1>>;
+    using Reg = RegTile<BaseTileRowMajor<Element>, tl::RowMajor<4, 8>>;
```
Comment (haruhi55) [inline tests/cpp/cell/test_s2r_copy.cu:223]: Could you please explain why lines 157, 161, 167, 187, 192, 220, and 222 have been commented out?

File: tests/cpp/cell/test_s2r_copy.cu [hunk 3]
```diff
@@ -236,13 +243,17 @@ TEST(TestShared2Reg, operand_A_swizzle) {
...
+    // const int kCols = 32;
+
+    const int kRows = 16;
+    const int kCols = 64;
 
     using SharedLayout = tl::RowMajor<kRows, kCols>;
     const bool kUseSwizzledLayout = true;
     using Shared = SharedTile<Element, SharedLayout, kUseSwizzledLayout>;
-    using Reg = RegTile<BaseTileRowMajor<Element>, tl::RowMajor<2, 2>>;
+    // using Reg = RegTile<BaseTileRowMajor<Element>, tl::RowMajor<2, 2>>;
```
Comment (haruhi55) [inline tests/cpp/cell/test_s2r_copy.cu:255]: Would combining lines 246–247 with line 255 produce a unit test case that's worth adding to the tests, or does it simply duplicate an existing test case?

File: tests/cpp/cell/test_s2r_copy.cu [hunk 4]
```diff
@@ -262,11 +273,16 @@ TEST(TestReg2Shared, operand_C_float) {
...
     const int kColRepeats = 8;
+    // const int kRowRepeats = 1;
+    // const int kColRepeats = 4;
     const int kRows = 16 * kRowRepeats;
     const int kCols = 16 * kColRepeats;
 
-    const int kWarpPerRow = 2;
-    const int kWarpPerCol = 2;
+    // const int kWarpPerRow = 2;
+    // const int kWarpPerCol = 2;
```
Comment (haruhi55) [inline tests/cpp/cell/test_s2r_copy.cu:282]: Do the test cases on lines 276–277 and 281–282 provide unique coverage worthy of being added to the tests, or do they simply duplicate existing tests?

If they offer unique coverage, please leave them in the comments; otherwise, they should be removed.

File: tests/cpp/cell/test_single_wmma.cu [hunk 5]
```diff
@@ -191,10 +193,12 @@ void run_test() {
 }
 
 TEST(TestWmma, test_m16n16k16_f) {
-    run_test<16, 16, 16>();  // Test the `BaseTile` 16x16x16
-    run_test<16, 32, 16>();
-    run_test<96, 48, 80>();
+    // run_test<16, 16, 16>();  // Test the `BaseTile` 16x16x16
+    // run_test<16, 32, 16>();
+    // run_test<96, 48, 80>();
```
Comment (haruhi55) [inline tests/cpp/cell/test_single_wmma.cu:198]: Please add a comment explaining why these test cases are commented out if they're intended to provide unique coverage but aren't supported yet. Otherwise, delete them if they're not needed.

File: include/cell/copy/warp.hpp (hunk 1 above)
Comment (KuangjuX) [reply include/cell/copy/warp.hpp]: Deleted useless codes.

File: tests/cpp/cell/test_s2r_copy.cu (hunk 3 above)
Comment (KuangjuX) [reply tests/cpp/cell/test_s2r_copy.cu]: Deleted useless codes.

File: tests/cpp/cell/test_s2r_copy.cu (hunk 4 above)
Comment (KuangjuX) [reply tests/cpp/cell/test_s2r_copy.cu]: Deleted useless codes.

File: tests/cpp/cell/test_single_wmma.cu (hunk 5 above)
Comment (KuangjuX) [reply tests/cpp/cell/test_single_wmma.cu]: Deleted useless codes.

File: tests/cpp/cell/test_s2r_copy.cu (hunk 2 above)
Comment (KuangjuX) [reply tests/cpp/cell/test_s2r_copy.cu:218]: Because I want to use the `WarpLayout` format which is more suitable for the current implementation.
